src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -740,6 +740,41 @@ void reorder_inputs::run(program& p, reorder_factory& rf) {
         }
     };
 
+    const auto reorder_eltwise = [&p](typed_program_node<eltwise>& eltwise_node) {
+        // Add reorder to align tensor size of eltwise inputs for NUMPY broadcasting
+        // This manual alignment of tensor size does not need if format is simple or compatible
+        auto& out_layout = eltwise_node.get_output_layout();
+        if (eltwise_node.need_input_tensors_size_align() && !format::is_simple_data_format(out_layout.format)) {
+            // Eltwise input tensor can be smaller than perferred format by NUMPY broad casting.
+            // e.g. (?,3,?,?,2) (?,?,2)
+            auto& pshape_a = eltwise_node.get_input_pshape(0);
+            auto& pshape_b = eltwise_node.get_input_pshape(1);
+            auto [ref_pshape, small_pshape, large_shape_idx, small_shape_idx] = (pshape_a.size() > pshape_b.size()) ?
+                                          std::make_tuple(pshape_a, pshape_b, 0, 1) : std::make_tuple(pshape_b, pshape_a, 1, 0);
+
+            auto const_dep_idx = eltwise_node.find_eltwise_const_dep_idx();
+            if (const_dep_idx.has_value() && const_dep_idx.value() < 2) {
+                auto const_input_pshape_size = eltwise_node.get_input_pshape(const_dep_idx.value()).size();
+                OPENVINO_ASSERT(const_input_pshape_size == ref_pshape.size() || const_input_pshape_size <= 1,
+                                "Unexpected pshape size of NUMPY broadcast of eltwise " + eltwise_node.id());
+            }
```

Comment:
Could you explain what it tries to check? I cannot understand why `constant` should be the larger one?